frontend/ui/screensaver.lua

```diff
@@ -157,6 +157,25 @@ function Screensaver:show()
                 width = Screen:getWidth(),
                 scale_factor = self:proportional() and 0 or nil,
             }
+        --fallback to random images if this book cover is excluded
+        else
+            local screensaver_dir = G_reader_settings:readSetting("screensaver_dir")
+            if screensaver_dir == nil then
```

Comment:
May be, just to avoid duplicating the screensaver_dir code, you could use just a bunch of standalone `if` instead of these `elseif`.
This would become:
```lua
        --fallback to random images if this book cover is excluded
        else
            screensaver_type = "random_images"
        end
        doc_settings:close()
    end
[...]
    if screensaver_type == "random_image" then
```
(This would just update the local `screensaver_type` , not the G_reader_setting one)

That way, you could have fallbacks for `bookstatus` or `readingprogress` : when no readerinstance, fallback to `screensaver_type = "message"` (which always work with the default message -  I think it would be better than disabled - Disabled being for those who really don't want any kind of screensaver).
And you could fallback too from random_image to message, if zero image in screensaver_dir.

Reply:
Really good idea.